src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/EditMode.java

```diff
@@ -48,12 +50,29 @@ private void submit() {
     driver.switchTo().defaultContent();
     wait.forElementClickable(submitButton);
     scrollAndClick(submitButton);
-    wait.forElementNotPresent(submitButtonBy);
-    PageObjectLogging.log("submit", "Page submitted", true);
+
+    PageObjectLogging.logInfo("Submit");
   }
 
   public ArticlePageObject submitArticle() {
     submit();
+    wait.forElementNotPresent(submitButtonBy);
+    PageObjectLogging.logInfo("Page submitted");
+
+    return new ArticlePageObject();
+  }
+
+  public ArticlePageObject submitArticleAsAnon() {
```

Comment:
Submitting as anon works this way only the first time we edit. Maybe go with "submitExpectingNotification" and then a regular submit. This way we won't hide test flow in test body ;)